cnd/src/swap_protocols/rfc003/ethereum/htlc_events.rs

```diff
@@ -104,25 +96,15 @@ async fn htlc_redeemed<A: Asset>(
     htlc_deployment: &Deployed<Ethereum>,
     start_of_swap: NaiveDateTime,
 ) -> anyhow::Result<Redeemed<Ethereum>> {
-    let pattern = TransactionPattern {
-        from_address: None,
-        to_address: None,
-        is_contract_creation: None,
-        transaction_data: None,
-        transaction_data_length: None,
-        events: Some(vec![Event {
-            address: Some(htlc_deployment.location),
-            data: None,
-            topics: vec![Some(Topic(*REDEEM_LOG_MSG))],
-        }]),
-    };
+    let events = vec![Event {
+        address: Some(htlc_deployment.location),
+        data: None,
```

Comment:
Do we ever use `data`?
Maybe you can remove the `Event` struct altogether and only pass the address and a list of topics to the function?

Reply:
I played with this a little and the conclusion I came to was we were trying to imitate web3 with the whole `Event` struct.  Is this assumption correct?

Reply:
Answering my own question; on further investigation we can remove this, update the comment that refers to web3 and still maintain clarity while reducing complexity - WIN.  Thanks

Reply:
I don't think we need to imitate something. The original design might have been inspired by that but the original design was also trying to be general purpose (which turned out to be a really bad idea :sweat_smile:).

Question everything my friend :)